### ISSUE TITLE
Hydration node mismatch in 3.1.0

### ISSUE THREAD

**Issue by cdwmhcc** [nuxt/nuxt#18491]
### Environment

------------------------------
- Operating System: `Windows_NT`
- Node Version:     `v19.2.0`
- Nuxt Version:     `3.1.0`
- Nitro Version:    `2.0.0`
- Package Manager:  `pnpm@7.25.1`
- Builder:          `vite`
- User Config:      `app`, `css`, `components`, `alias`, `modules`, `colorMode`
- Runtime Modules:  `@nuxtjs/color-mode@3.2.0`, `@pinia/nuxt@0.4.6`
- Build Modules:    `-`
------------------------------

### Reproduction

[codesandbox](https://codesandbox.io/p/sandbox/adoring-dirac-pbw7ds?selection=%5B%7B%22endColumn%22%3A1%2C%22endLineNumber%22%3A6%2C%22startColumn%22%3A1%2C%22startLineNumber%22%3A6%7D%5D&file=%2Fpages%2Findex.vue)

### Describe the bug

```
[Vue warn]: Hydration children mismatch in <div>: server rendered element contains fewer child nodes than client vdom. 
  at <Spin> 
  at <Index onVnodeUnmounted=fn<onVnodeUnmounted> ref=Ref< undefined > > 
  at <RouteProvider key="/" routeProps= {Component: {…}, route: {…}} pageKey="/"  ... > 
  at <FragmentWrapper > 
  at <RouterView name=undefined route=undefined > 
  at <NuxtPage> 
  at <App key=2 > 
  at <NuxtRoot>
```

**Comment by cdwmhcc** [comment]
Everything works fine in 3.0.0.

**Comment by manniL** [comment]
Seems like the default slot is not rendered on the server-side.

Client
<img width="151" alt="image" src="https://user-images.githubusercontent.com/640208/214395532-bba8daac-17d8-49ed-87c2-5e745bd7256a.png">

Server
<img width="562" alt="image" src="https://user-images.githubusercontent.com/640208/214395584-42f168ed-be53-4090-88ce-5df5badd5fe3.png">

**Comment by madebyfabian** [comment]
I can confirm this on nuxt 3.1.0. Also getting a hydration mismatch error. But only when using a method in the script. So

Works fine:
```vue
<template>
	<div>This component will only be rendered on the server.</div>
</template>

<script setup lang="ts">
	console.log('Server only!')
</script>
```

Also works fine:
```
// utils/example.ts
export const example = () => { return true }
```
```vue
<template>
	<div>This component will only be rendered on the server.</div>
</template>

<script setup lang="ts">
	const exampleRes = example()
	console.log('Server only!', { exampleRes })
</script>

```

Hydration mismatch (e.g. using another packages method, e.g. `nuxt-graphql-client`'s `useAsyncGql` method).
```vue
<template>
	<div>This component will only be rendered on the server.</div>
</template>

<script setup lang="ts">
	const { data, error } = await useAsyncGql({
		operation: 'ListPosts',
	})
	console.log('Server only!', { data: data.value, error: error.value })
</script>
```

So it seems that there are issues in server-components when using composables from other packages. The `useAsyncGql` composable should work here because it is one that runs on the server + client normally.

**Comment by danielroe** [comment]
@madebyfabian I believe your issue has a different cause. Would you create a new issue to track?

**Comment by madebyfabian** [comment]
@danielroe Sure! https://github.com/nuxt/nuxt/issues/18500

**Comment by danielroe** [comment]
This issue has been resolved in https://github.com/nuxt/nuxt/pull/18494. I'd appreciate testing in the [edge channel](https://nuxt.com/docs/guide/going-further/edge-channel/), as I plan to make a patch release shortly to include this fix.

**Comment by MiniDigger** [comment]
> This issue has been resolved in #18494. I'd appreciate testing in the [edge channel](https://nuxt.com/docs/guide/going-further/edge-channel/), as I plan to make a patch release shortly to include this fix.

works fine on edge for me
